src/components/Swap/Toolbar/index.tsx

```diff
@@ -186,7 +175,20 @@ export default memo(function Toolbar() {
   return (
     <StyledExpando
       title={
-        <ToolbarRow flex justify="space-between" data-testid="toolbar">
+        <ToolbarRow
+          flex
+          justify="space-between"
+          data-testid="toolbar"
+          onClick={(e) => {
+            // e.currentTarget is ToolbarRow, the element this listener is attached to.
+            // e.target is the element that was clicked, which could be ToolbarRow or a child.
+            // The user interaction should be handled by the child.
+            if (e.currentTarget !== e.target) {
```

Comment:
I'd prefer that children that alter the click behavior also call e.stopPropagation, instead of having the parent handler here check for child targeting.

I can see how either could be argued for, but this feels more confusing / less idiomatic to me. I also think that it will be harder to debug, as I tend to start looking for event handlers on the child and bubbling up (following the DOM model) - so if something goes wrong it will be harder to pinpoint it to a parents event handler.